number.js

```diff
@@ -1,106 +1,106 @@
-     *
-     * See http://en.wikipedia.org/wiki/Continued_fraction for implementation
-     * details
-     *
-     * toFraction(4/8) => [1, 2]
-     * toFraction(0.66) => [33, 50]
-     * toFraction(0.66, 0.01) => [2/3]
-     * toFraction(283 + 1/3) => [850, 3]
-     */
-    toFraction: function(decimal, tolerance, max_denominator) {
-        max_denominator = max_denominator || 1000;
-        tolerance = tolerance || EPSILON; // can't be 0
-
-        // Initialize everything to compute successive terms of
-        // continued-fraction approximations via recurrence relation
-        var n = [1, 0], d = [0, 1];
-        var a = Math.floor(decimal), t;
-        var rem = decimal - a;
-
-        while (d[0] <= max_denominator) {
-            if (knumber.equal(n[0] / d[0], decimal, tolerance)) {
-                return [n[0], d[0]];
-            }
-            n = [a*n[0] + n[1], n[0]];
-            d = [a*d[0] + d[1], d[0]];
-            a = Math.floor(1 / rem);
-            rem = 1/rem - a;
+export function equal(x: number, y: number, tolerance?: number): boolean {
+    // Checking for undefined makes this function behave nicely
+    // with vectors of different lengths that are _.zip'd together
+    if (x == null || y == null) {
+        return x === y;
+    }
+    // We check === here so that +/-Infinity comparisons work correctly
+    if (x === y) {
+        return true;
+    }
+    if (tolerance == null) {
+        tolerance = DEFAULT_TOLERANCE;
+    }
+    return Math.abs(x - y) < tolerance;
+}
+
+export function sign(
+    x: number,
+    tolerance?: number
+): number /* Should be: 0 | 1 | -1 */ {
+    return equal(x, 0, tolerance) ? 0 : Math.abs(x) / x;
+}
+
+export function isInteger(num: number, tolerance?: number): boolean {
+    return equal(Math.round(num), num, tolerance);
+}
+
+// Round a number to a certain number of decimal places
+export function round(num: number, precision: number): number {
+    var factor = Math.pow(10, precision);
+    return Math.round(num * factor) / factor;
+}
+
+// Round num to the nearest multiple of increment
+// i.e. roundTo(83, 5) -> 85
+export function roundTo(num: number, increment: number): number {
+    return Math.round(num / increment) * increment;
+}
+
+export function floorTo(num: number, increment: number): number {
+    return Math.floor(num / increment) * increment;
+}
+
+export function ceilTo(num: number, increment: number): number {
+    return Math.ceil(num / increment) * increment;
+}
+
+/**
+ * toFraction
+ *
+ * Returns a [numerator, denominator] array rational representation
+ * of `decimal`
+ *
+ * See http://en.wikipedia.org/wiki/Continued_fraction for implementation
+ * details
+ *
+ * toFraction(4/8) => [1, 2]
+ * toFraction(0.66) => [33, 50]
+ * toFraction(0.66, 0.01) => [2/3]
+ * toFraction(283 + 1/3) => [850, 3]
+ */
+export function toFraction(
+    decimal: number,
+    tolerance: number = EPSILON, // can't be 0
+    max_denominator: number = 1000
+): [number, number] {
```

Comment:
I wonder if we should introduce a `Fraction` type.

Reply:
We could, but I don't know that it makes things much clearer. Ultimately, I wonder if we'd be better off with an object type like: `{| numerator: number, denominator: number |}` to make it a lot clearer both here and all the places we use it in webapp.